<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>XVideoUIClass</class>
 <widget class="QWidget" name="XVideoUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>920</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>XVideoUI</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton:hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(120, 120, 120, 255), stop:0.495 rgba(39, 39, 39, 255), stop:0.505 rgba(39,39, 39, 255), stop:1 rgba(120, 120,120, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }
QPushButton:!hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(39, 39, 39, 255), stop:0.495 rgba(120, 120, 120, 255), stop:0.505 rgba(120,120, 120, 255), stop:1 rgba(39, 39,39, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }

#XVideoUIClass
{
	background-color:rgb(76, 58, 55);
}
#closeButton
{
	background-color: rgb(53, 53, 53);
	color: rgb(255, 255, 255);
	font: 75 11pt &quot;等线&quot;;
}

QLabel
{
	color:rgb(255, 255, 255);
	
	font: 10pt &quot;黑体&quot;;
}
#playButton{
	
	background-color: rgba(255, 255, 255, 0);
}
#playButton:!hover{
	image: url(:/XVideoUI/Resources/PlayN.png);
	background-color: rgba(255, 255, 255, 0);
}
#playButton:hover{
	image: url(:/XVideoUI/Resources/PlayH.png);
	background-color: rgba(255, 255, 255, 0);
}
#playButton:pressed{
	image: url(:/XVideoUI/Resources/PlayP.png);
	background-color: rgba(255, 255, 255, 0);
}

#pauseButton:!hover{
	image: url(:/XVideoUI/Resources/PauseN.png);
	background-color: rgba(255, 255, 255, 0);
}
#pauseButton:hover{
	image: url(:/XVideoUI/Resources/PauseH.png);
	background-color: rgba(255, 255, 255, 0);
}
#pauseButton:pressed{
	image: url(:/XVideoUI/ResourcesPauseP.png);
	background-color: rgba(255, 255, 255, 0);
}
</string>
  </property>
  <widget class="QPushButton" name="closeButton">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>10</y>
     <width>31</width>
     <height>31</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string>X</string>
   </property>
   <property name="flat">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="XVideoWidget" name="src1video">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>30</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="openButton">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>320</y>
     <width>61</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
  </widget>
  <widget class="QSlider" name="playSlider">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>320</y>
     <width>381</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>360</y>
     <width>81</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>亮度[0~100]</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>620</x>
     <y>400</y>
     <width>111</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>对比度[1.0~3.0]</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="bright">
   <property name="geometry">
    <rect>
     <x>750</x>
     <y>370</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="contrast">
   <property name="geometry">
    <rect>
     <x>750</x>
     <y>410</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>1.000000000000000</double>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="setButton">
   <property name="geometry">
    <rect>
     <x>830</x>
     <y>360</y>
     <width>51</width>
     <height>51</height>
    </rect>
   </property>
   <property name="text">
    <string>设置</string>
   </property>
  </widget>
  <widget class="XVideoWidget" name="des">
   <property name="geometry">
    <rect>
     <x>490</x>
     <y>30</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="exportButton">
   <property name="geometry">
    <rect>
     <x>770</x>
     <y>320</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>导出</string>
   </property>
  </widget>
  <widget class="QPushButton" name="playButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>340</y>
     <width>51</width>
     <height>61</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="pauseButton">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>340</y>
     <width>51</width>
     <height>61</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QComboBox" name="rotate">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>450</y>
     <width>101</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>0</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>90</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>180</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>270</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>440</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>图像旋转</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>480</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>图像镜像</string>
   </property>
  </widget>
  <widget class="QComboBox" name="flip">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>490</y>
     <width>101</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>不处理</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>左右镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下左右镜像</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>540</y>
     <width>81</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>图像尺寸W/H</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="width">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>540</y>
     <width>111</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>99999.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="height">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>560</y>
     <width>111</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>9999.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>540</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>金字塔</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pydown">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>540</y>
     <width>111</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>100.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pyup">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>560</y>
     <width>111</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>100.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>480</y>
     <width>61</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图像裁剪</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="cx">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>480</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>999.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="cy">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>480</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>999.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="cw">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>500</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>999.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ch">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>500</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>999.000000000000000</double>
   </property>
  </widget>
  <widget class="QComboBox" name="color">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>440</y>
     <width>101</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>RGB图像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>灰度图像</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>500</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>(x,y,w,h)</string>
   </property>
  </widget>
  <widget class="QPushButton" name="markButton">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>600</y>
     <width>61</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>水印</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="mx">
   <property name="geometry">
    <rect>
     <x>730</x>
     <y>610</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>999.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="my">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>610</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>999.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ma">
   <property name="geometry">
    <rect>
     <x>830</x>
     <y>610</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="XVideoWidget" name="src2video">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>400</y>
     <width>380</width>
     <height>280</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="blendButton">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>610</y>
     <width>61</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>融合</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ba">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>650</y>
     <width>51</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="value">
    <double>0.500000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="mergeButton">
   <property name="geometry">
    <rect>
     <x>660</x>
     <y>640</y>
     <width>61</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>合并</string>
   </property>
  </widget>
  <widget class="QSlider" name="left">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>350</y>
     <width>381</width>
     <height>22</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal{
	background:rgb(0,170,255);
	border-radius:5px;
	background-color: rgb(237, 32, 255);
}</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QSlider" name="right">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>370</y>
     <width>381</width>
     <height>22</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal{
	background:rgb(0,170,255);
	border-radius:5px;
	background-color: rgb(237, 32, 255);
}</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="value">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>XVideoWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>xvideowidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="XVideoUI.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>752</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>641</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>432</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderPressed()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SliderPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SliderRelease()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>281</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SetPos(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>213</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Set()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>619</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>619</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exportButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Export()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>516</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>270</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Pause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>350</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>markButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Mark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>618</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>-57</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blendButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Blend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>717</y>
    </hint>
    <hint type="destinationlabel">
     <x>63</x>
     <y>-30</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mergeButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Merge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>709</y>
    </hint>
    <hint type="destinationlabel">
     <x>213</x>
     <y>-30</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>left</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Left(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>29</x>
     <y>358</y>
    </hint>
    <hint type="destinationlabel">
     <x>39</x>
     <y>-84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Right(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>82</x>
     <y>-22</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Open()</slot>
  <slot>SliderPress()</slot>
  <slot>SliderRelease()</slot>
  <slot>SetPos(int)</slot>
  <slot>Set()</slot>
  <slot>Export()</slot>
  <slot>Play()</slot>
  <slot>Pause()</slot>
  <slot>Mark()</slot>
  <slot>Blend()</slot>
  <slot>Merge()</slot>
  <slot>Left(int)</slot>
  <slot>Right(int)</slot>
 </slots>
</ui>
